### PR TITLE
change test to ensure use of /

### LINKED ISSUE #11986: Divide one Decimal by Another with JavaScript
Challenge [Divide one Decimal by Another with JavaScript](https://www.freecodecamp.com/challenges/divide-one-decimal-by-another-with-javascript#?solution=%0Avar%20quotient%20%3D%200.0%20%2F%202.0%3B%0A%0A%0A%0A%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (Windows NT 10.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.75 Safari/537.36</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

My code:
```javascript

var quotient = 0.0 / 2.0;


```

Problem is solved by leaving the original code as is and adding:

```javascript
quotient = 2.2;
```

I believe this should not be a valid answer, unless this is intended?

### PR DESCRIPTION
**Pull request by dhcodes**
#### Pre-Submission Checklist

- [X] Your pull request targets the `staging` branch of FreeCodeCamp.
- [X] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [X] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [X] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [X] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [X] Tested changes locally.
- [X] Closes currently open issue (replace XXXX with an issue no): Closes #11986 

#### Description

Changed the regex test to be more strict.

### CODE REVIEW

File: seed/challenges/01-front-end-development-certification/basic-javascript.json [hunk 1]
```diff
@@ -691,7 +691,7 @@
       ],
       "tests": [
         "assert(quotient === 2.2, 'message: The variable <code>quotient</code> should equal <code>2.2</code>.');",
-        "assert(/\\//.test(code), 'message: You should use the <code>/</code> operator');"
+        "assert(/4\\.4\\s*\\/\\s*2\\.0/.test(code), 'message: You should use the <code>/</code> operator');"
```
Comment (erictleung) [inline seed/challenges/01-front-end-development-certification/basic-javascript.json]: Hypothetically, I could write `4.400 / 2.0` and it will fail this new test. Other ways we could prevent issues like this is maybe having a regex count the number of `quotient` variables to make sure there is only one? And also maybe have a comment in the code like this

```js
var quotient = 0.0 / 2.0; // Fix this line
```

Comment (erictleung) [inline seed/challenges/01-front-end-development-certification/basic-javascript.json]: The reason I say to have something to count `quotient` variables, although a work around, is that right now, if you have just `var quotient = 2.2;`, this will fail because there is no division sign (which is the second test).

But then if they did something like the original issue has, of 

```js
var quotient = 0.0 / 2.0;
quotient = 2.2;
```

then the test I propose of counting that there is only one `quotient` variable will catch this.

So what I'm proposing is to have like three tests, the current two and another one counting the number of `quotient` variables and the new test would say something like `Change the numbers on the right side of the equal side` or maybe `There should only be one mention of the quotient variable` (this one might be confusing).

Another edge case I thought about is if they did some math like `quotient = 1.1 + 1.1;` after the initial `var quotient = 0.0 / 2.0;` which wouldn't be caught by a regex.

I hope my reasoning makes sense. In sum, I'm trying to catch as many edge cases as possible with any new tests. Let me know if something isn't clear.

Comment (dhcodes) [inline seed/challenges/01-front-end-development-certification/basic-javascript.json]: I agree and that makes sense. Better to make the tests as bulletproof as possible with each update. I'll take a crack at this later today.

**Comment by dhcodes** [comment]
@erictleung finally got around to updating this. I shouldn't put a time estimate lol 😆 . Let me know what you think.

**Comment by dhcodes** [comment]
put this on hold again. The `npm run test-challenges` is having an issue.

File: seed/challenges/01-front-end-development-certification/basic-javascript.json [hunk 2]
```diff
@@ -691,7 +691,9 @@
       ],
       "tests": [
         "assert(quotient === 2.2, 'message: The variable <code>quotient</code> should equal <code>2.2</code>.');",
-        "assert(/\\//.test(code), 'message: You should use the <code>/</code> operator');"
+        "assert(/4\\.40*\\s*\\/\\s*2\\.*0*/.test(code), 'message: You should use the <code>/</code> operator.');",
+        "assert(code.match(/quotient/g).length === 3, 'message: The quotient variable should only be assigned once.');"
```
Comment (erictleung) [inline seed/challenges/01-front-end-development-certification/basic-javascript.json]: TravisCI might be angry at this line. This is checking for 3 instances of `quotient`, but the solution should only have 1 instance of `quotient`.

image

So the test should be `code.match(/quotient/g).length === 1`, unless I'm missing something.

**Review by erictleung — DISMISSED** [review]

File: seed/challenges/01-front-end-development-certification/basic-javascript.json (hunk 2 above)
Comment (dhcodes) [inline seed/challenges/01-front-end-development-certification/basic-javascript.json]: @erictleung Yeah, but this is caused by the code.match Regex tests now including the tail. So my only option are to remove the suggested solution from the json or remove the tail. I'm going to open an issue referencing this.

**Comment by BerkeleyTrue** [comment]
@dhcodes How is this pr coming along?

**Comment by dhcodes** [comment]
@BerkeleyTrue It's kind of in limbo because the length value in the line `assert(code.match(/quotient/g).length === 3` is false for the solution (therefore doesn't pass) but is true when incorporating the code in the tail (which the tests now do). So I was asking whether to remove the solution code and rely on the tests or to remove the tail. I think the former but wanted feedback.

File: seed/challenges/01-front-end-development-certification/basic-javascript.json (hunk 2 above)
Comment (dhcodes) [inline seed/challenges/01-front-end-development-certification/basic-javascript.json]: @erictleung I've confirmed with Berkeley that it's alright to delete the solution. This should be good to go now.

**Comment by dhcodes** [comment]
@freeCodeCamp/moderators ready for QA and merge

File: seed/challenges/02-javascript-algorithms-and-data-structures/basic-javascript.json [hunk 3]
```diff
@@ -679,19 +679,19 @@
...
         "(function(y){return 'quotient = '+y;})(quotient);"
       ],
-      "solutions": [
-        "var quotient = 4.4 / 2.0;"
-      ],
+      "solutions": [],
       "tests": [
         "assert(quotient === 2.2, 'message: The variable <code>quotient</code> should equal <code>2.2</code>.');",
-        "assert(/\\//.test(code), 'message: You should use the <code>/</code> operator');"
+        "assert(/4\\.40*\\s*\\/\\s*2\\.*0*/.test(code), 'message: You should use the <code>/</code> operator.');",
```
Comment (systimotic) [inline seed/challenges/02-javascript-algorithms-and-data-structures/basic-javascript.json]: This regex is really close to perfect! 
`4.4 / 20` will also pass this test. It's not a big deal, as it will break the first test, so one would have to do something crazy like 
`(4.4 / 20 * 0) + 2.2` is also a way pass, but I feel like if a camper can come up with that, they understand.
Maybe we should make the test more lenient? Something like `/\d\s*\/\s*\d/` should do the trick, while allowing weird (but correct) solutions like `8.8 / 4` (which should pass, according to the message).

Comment (dhcodes) [inline seed/challenges/02-javascript-algorithms-and-data-structures/basic-javascript.json]: @systimotic last reviewer said to make it more strict lol.

Comment (systimotic) [inline seed/challenges/02-javascript-algorithms-and-data-structures/basic-javascript.json]: @dhcodes I'm ok with that too, but in that case I think the message should be updated to reflect what the tests are actually testing for.

Comment (dhcodes) [inline seed/challenges/02-javascript-algorithms-and-data-structures/basic-javascript.json]: @systimotic I'll need your take on what that should be.

Comment (systimotic) [inline seed/challenges/02-javascript-algorithms-and-data-structures/basic-javascript.json]: @dhcodes How does "You should use the `/` operator to divide 4.4 by 2" sound?

Comment (dhcodes) [inline seed/challenges/02-javascript-algorithms-and-data-structures/basic-javascript.json]: Like solid gold. 🏅

**Comment by dhcodes** [comment]
Attempt no. 563 ready for your review @systimotic

**Review by systimotic — CHANGES_REQUESTED** [review]
Lol imma be mean 😅 😢

I can't even blame you, because it was inconsistent before you started making changes. So I looked at a few challenges below this one, and none of their tests end with a period. Your new test does. Can you remove the periods?
Soooooooo close 😂

**Comment by dhcodes** [comment]
@systimotic Attempt no. 564

**Review by systimotic — APPROVED** [review]
```
This is TimoCI speaking.

TimoCI approves of attempt no. 564
```
😄 👍 👍 👍  **LGTM**
It's finally done! Only took us 40 days... 😅
Thanks for not giving up @dhcodes 🙂
